src/pages/Submission.js

```diff
@@ -25,7 +25,16 @@ export default () => {
     const submittedProjectRef = d.submittedProject
     if (!!submittedProjectRef) {
       const submission = await getSubmission(submittedProjectRef)
-      setSubmission(submission)
+
+      Object.keys(submission.grades).forEach(id => {
+        if (submission.grades[id].removed) {
+          delete submission.grades[id]
+        }
+      })
```

Comment:
why not filter here?

Reply:
I think we need to keep it as an object not an array

Reply:
ah forgot this was an object, ok